docs/debugging.md

```diff
@@ -0,0 +1,135 @@
+# Debugging
+
+For basics on how to set-up your repo and how to build the product, refer to the [contributing guide](../CONTRIBUTING.md).
+
+## Debugging and testing NuGet.exe (NuGet.Commandline)
+
+Given that it is a .NET Framework based x86 console application, NuGet.exe is straightforward to debug.
+
+Set [NuGet.CommandLine](../src/NuGet.Clients/NuGet.CommandLine/NuGet.CommandLine.csproj) as the start-up project in the [NuGet.sln](../NuGet.sln), and set the parameters in Debug window for the project.
+
+### Testing the build of NuGet.exe
+
+Alternatively, if you build the the repository under a `Debug` configuration, which is the default, you can simply pass a `--debug` argument to NuGet.exe when invoking it to get a debugger prompt.
+
+### Code pointers for NuGet.exe
+
+* The entry point for all commands is [Program.cs](../src/NuGet.Clients/NuGet.CommandLine/Program.cs) in the same project.
+* Each command in NuGet.exe has it's own class. For example if you want to debug restore, you can set a breakpoint in [RestoreCommand.cs](../src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs). Note that there are multiple instances of the RestoreCommand type.
+
+## Debugging and testing NuGet in Visual Studio
+
+Testing the NuGet Visual Studio functionality is equally as easy as testing NuGet.exe.
+The start-up project is [NuGet.VisualStudio.Client](../src/NuGet.Clients/NuGet.VisualStudio.Client/NuGet.VisualStudio.Client.csproj). Starting this will launch the [experimental instance](https://docs.microsoft.com/en-us/visualstudio/extensibility/the-experimental-instance) of Visual Studio.
+
+### Testing the build of NuGet in Visual Studio
+
+NuGet functions as an extension on top of Visual Studio.
+However NuGet is also considered a system component which means it cannot be managed in the same way as other Visual Studio extensions. However, it is possible to install a locally built NuGet in a Visual Studio instance.
+
+#### Installing the NuGet extension in Visual Studio
```

Comment:
```suggestion
#### Installing a custom version of the NuGet extension in Visual Studio
```

Reply:
This requires some link changes. I'll make sure they are addressed